### PR TITLE
feat: ForkChoiceNotifications for BlockchainProvider and NoopProvider

### PR DESCRIPTION
**Pull request by royvardhan**
Ref: https://github.com/paradigmxyz/reth/issues/9521

### CODE REVIEW

File: crates/chain-state/src/chain_info.rs [hunk 1]
```diff
@@ -123,6 +123,16 @@ impl ChainInfoTracker {
...
         });
     }
+
+    /// Subscribe to the finalized block.
+    pub fn subscribe_to_finalized_block(&self) -> watch::Receiver<Option<SealedHeader>> {
+        self.inner.finalized_block.subscribe()
+    }
+
+    /// Subscribe to the safe block.
+    pub fn subscribe_to_fork_choice(&self) -> watch::Receiver<Option<SealedHeader>> {
```
Comment (mattsse) [inline crates/chain-state/src/chain_info.rs:133]:
```suggestion
    pub fn subscribe_to_safe_block(&self) -> watch::Receiver<Option<SealedHeader>> {
```

File: crates/storage/provider/src/providers/mod.rs [hunk 2]
```diff
@@ -905,6 +905,16 @@ where
     }
 }
 
+impl<DB> ForkChoiceSubscriptions for BlockchainProvider<DB>
+where
+    DB: Send + Sync,
+{
+    fn subscribe_to_fork_choice(&self) -> ForkChoiceNotifications {
```
Comment (mattsse) [inline crates/storage/provider/src/providers/mod.rs:912]: we should actually rename this to safe and and finalized

**Review by mattsse — CHANGES_REQUESTED** [review]
last naming nits

**Comment by royvardhan** [comment]
Looks good? @mattsse

**Review by mattsse — APPROVED** [review]
nice work

**Pull request merged**